MID Server/MID Server Script Includes/IllumioHTTPClient.js

```diff
@@ -105,15 +107,22 @@ IllumioHTTPClient.prototype = {
                 };
 
                 return responseObj;
-            } else {
-
+            }
+            if (retryCount == 0 || (RETRY_CODES.indexOf(responseCode) == -1 && !(responseCode >= 500))) {
                 return {
                     hasError: true,
                     status: responseCode
                 };
             }
+
+            lastSleepTime = lastSleepTime ? lastSleepTime : 0;
+            var currentSleepTime = Math.min(this.HTTP_RETRY_INTERVAL_MAX, lastSleepTime + this.HTTP_RETRY_INTERVAL_INCREMENT);
```

Comment:
One consideration here - some responses (particularly 429s) return a `Retry-After` header that should supercede the `retryInterval` value. We should check for this header and use it if it exists and is non-zero.

Reply:
Sure, can do this